f600233d1b154d507b782f84604bcb12/update/sys_hub_action_type_definition_82657a551b2111107c458623604bcb46.xml

```diff
@@ -138,252 +138,324 @@
-                          "itemDescriptionOrConfigSummary": configSummary,
-                          "monthlyPrice": instance.monthly_price,
-                          "monthsInPeriod": monthsInPeriod(period.periodUnit, period.periodUnitCount),
-                      });
-                  });
-              }
-              else if (instance.need_for_entire_task_order_duration) {
-                  periods_estimate.forEach(function (period) {
-                      period.periodLineItems.push({
-                          "clin": "x001",
-                          "idiqClin": getClinFromDowTask(instance.dow_task_number, outputs.new_clins),
-                          "dowTaskNumber": instance.dow_task_number,
-                          "serviceOffering": item.service_offering.name,
-                          "itemDescriptionOrConfigSummary": configSummary,
-                          "monthlyPrice": instance.monthly_price,
-                          // when entire period need to go in another layer to get period info
-                          "monthsInPeriod": monthsInPeriod(period.period.periodUnit, period.period.periodUnitCount),
-                      });
-                  });
-              }
-          });
-      }
-  });
 
-  // Retrieve surge_capabilities from requirements cost estimate table
-  if (requirementsCostEstimate) {
-      var gr = new GlideRecord('x_g_dis_atat_requirements_cost_estimate');
-      gr.get(requirementsCostEstimate);
-      outputs.surge_capabilities = gr.surge_capabilities;
-  }
+    // For each item in selected_service_offerings
+    outputs.selected_service_offerings.forEach(function (item) {
+
+        // If the item includes classification_instances
+        if (item.classification_instances) {
+            // For each instance in classification instances array
+            item.classification_instances.forEach(function (instance) {
+
+                // Check if need for entire task order duration is false
+                // If false, use the selected_periods array of the classification instance
+                if (!instance.need_for_entire_task_order_duration) {
+
+                    // For each period in the selected_periods array of the classification instance
+                    instance.selected_periods.forEach(function (period) {
+
+                        // Map period_type and base_period of the periods_estimate array
+                        var formattedPeriodOutput = periods_estimate.map(function (input) {
+                            return input.period.periodType + input.period.optionOrder;
+                        });
+
+                        var indexOfCurrentPeriod = formattedPeriodOutput.indexOf(period.periodType + period.optionOrder);
+                        // Insert into periods_estimate, under the specified period and period_line_items
+                        periods_estimate[indexOfCurrentPeriod].periodLineItems.push({
+                            "clin": "x001",
```

Comment:
Is this intentionally hard-coded?

Reply:
I think we have a plan in the future tickets for these CLINs, because as-is they are calculated on the backend (in the lambda) when they are now required for the frontend.